docs/tutorial_email_generator.md

```diff
@@ -0,0 +1,225 @@
+
+```python
+import openai
+openai.api_key = 'YOUR_OPENAI_API_KEY'
+
+def generate_email(subject, recipient_name, content_description):
+    prompt = (
+        f"""Compose an email to {recipient_name} with the subject '{subject}'.
+        The email should cover the following points: {content_description}.
+        Keep it concise.
+        """
+    )
+    response = openai.ChatCompletion.create(model='gpt-3.5-turbo',
+                                            messages=[{
+                                                "role": "system",
+                                                "content": prompt
+                                            }],
+                                            max_tokens=500
+                                            )
+    email_body = response['choices'][0]['message']['content'].strip()
+    return email_body
+```
+
+Now, let's try it out on a couple of examples.
+```python
+subject = "Japan trip"
+recipient_name = "Alex"
+content_description = """
+- Went to Tokyo
+- Visited Tokyo Sky Tree
+- Favorite part was the food, ate lots of ramen
+"""
+email = generate_email(subject, recipient_name, content_description)
+print(email)
+
+subject = "Running late for the 10am meeting!"
+recipient_name = "Kenny"
+content_description = """
+- will be 10mins late for the 10am meeting
+- messed up a train transfer
+- say that I'm sorry
+"""
+email = generate_email(subject, recipient_name, content_description)
+print(email)
+```
+
+Below are the app's outputs on those two examples.
+```text
+Subject: Japan trip
+
+Hey Alex,
+
+Just wanted to fill you in on my Japan trip. I went to Tokyo and it was amazing!
+
+One highlight was visiting the Tokyo Sky Tree. The views were absolutely incredible!
+
+But my absolute favorite part was the food. I ate so much ramen, it was delicious!
+
+Catch up soon,
+
+[Your Name]
+```
+```text
+Subject: Running late for the 10am meeting!
+
+Hi Kenny,
+
+I just wanted to drop you a quick email to let you know that I will be about 10 minutes late for the 10am meeting this morning. I messed up a train transfer and it set me back a bit. I apologize for any inconvenience caused.
+
+See you shortly.
+
+Best,
+[Your Name]
+```
+
+Based on these couple examples, it seems like the app is doing a pretty good job. However, it's not always perfect - in the example below, the app "hallucinates" a bunch of fake qualifications that we did not specify!
+
+```python
+subject = "Application for the software engineer role"
+recipient_name = "Google"
+content_description = """
+- I want to apply for the role
+- I have many years of experience in AI, making me a strong candidate
+- Ready to start ASAP
+"""
+email = generate_email(subject, recipient_name, content_description)
+print(email)
+```
+```text
```

Comment:
I think code blocks don't do text wrapping, which makes this a bit hard to read.

Maybe we can use a quote block for this or something?

https://myst-parser.readthedocs.io/en/latest/syntax/typography.html#quotations

Reply:
hmm I actually kind of liked how it doesn't do text wrapping, since then it takes up much less screen real estate. I also didn't think it was that important for the email to be easy to read, since the exact contents of the emails aren't really that important 😅 Here's how it looks with quote blocks though vs. the code text block from before. WDYT?

<img width="1064" alt="image" src="https://github.com/citadel-ai/langcheck/assets/107823399/081ed2e6-7546-4c21-a5fc-d121977bd5f4">

<img width="1062" alt="image" src="https://github.com/citadel-ai/langcheck/assets/107823399/0fe78486-270e-4386-bae4-910e7b17fb5e">

Reply:
I think a first time reader will want to identify the hallucinations in the email that you pointed out, which was kind of hard with scrolling. It's pretty easy to visually skip the block if someone's not interested in reading. 

(We could also edit the email a bit so it's not as long)

Reply:
Got it - changed everything to use quote blocks

Reply:
Thanks!